python/pyspark/mllib/clustering.py

```diff
@@ -466,7 +557,8 @@ def predictOnValues(self, dstream):
 
 def _test():
     import doctest
-    globs = globals().copy()
+    import pyspark.mllib.clustering
+    globs = pyspark.mllib.clustering.__dict__.copy()
```

Comment:
Yes, the original code will raise "Py4JError: Trying to call a package" at https://github.com/apache/spark/pull/6992/files#diff-21c55b407050d37f67a2919470e047ebR324 .
I took a glance over related code and found in https://github.com/apache/spark/blob/master/python/pyspark/mllib/recommendation.py#L209 it use __dict__copy(), so I try the same method here and it works well.
Then I investigate the difference between the two ways.
Running the following code:
```python
globs1 = globals().copy()
globs2 = pyspark.mllib.clustering.__dict__.copy()

print [k for k in glob1 if k not in glob2]
print [k for k in glob2 if k not in glob1] 
``` 
and the output is:
```
['__loader__']
[]
```
It means that globs1(can't pass test) has one elements called ```__loader__``` more than globs2(can pass). I have no idea of what cause this.